src/main/java/de/tum/in/www1/artemis/web/rest/RatingResource.java

```diff
@@ -0,0 +1,126 @@
+        this.ratingService = ratingService;
+        this.userService = userService;
+        this.authCheckService = authCheckService;
+        this.resultService = resultService;
+        this.participationService = participationService;
+    }
+
+    /**
+     * Return Rating referencing resultId or null
+     *
+     * @param resultId - Id of result that is referenced with the rating
+     * @return Rating or null
+     */
+    @GetMapping("/results/{resultId}/rating")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Optional<Rating>> getRatingForResult(@PathVariable Long resultId) {
+        // TODO allow for Instructors
+        if (!checkIfUserIsOwnerOfSubmission(resultId) && !authCheckService.isAdmin()) {
+            return forbidden();
+        }
+        Optional<Rating> rating = ratingService.findRatingByResultId(resultId);
+        return ResponseEntity.ok(rating);
+    }
+
+    /**
+     * Persist a new Rating
+     *
+     * @param resultId    - Id of result that is referenced with the rating that should be persisted
+     * @param ratingValue - Value of the updated rating
+     * @return inserted Rating
+     * @throws URISyntaxException if the Location URI syntax is incorrect
+     */
+    @PostMapping("/results/{resultId}/rating/{ratingValue}")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Rating> createRatingForResult(@PathVariable Long resultId, @PathVariable Integer ratingValue) throws URISyntaxException {
+        if (!checkIfUserIsOwnerOfSubmission(resultId)) {
+            return forbidden();
+        }
+
+        Rating savedRating = ratingService.saveRating(resultId, ratingValue);
+        return ResponseEntity.created(new URI("/api/results/" + savedRating.getId() + "/rating")).body(savedRating);
+    }
+
+    /**
+     * Update a Rating
+     *
+     * @param resultId    - Id of result that is referenced with the rating that should be updated
+     * @param ratingValue - Value of the updated rating
+     * @return updated Rating
+     */
+    @PutMapping("/results/{resultId}/rating/{ratingValue}")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Rating> updateRatingForResult(@PathVariable Long resultId, @PathVariable Integer ratingValue) {
+        if (!checkIfUserIsOwnerOfSubmission(resultId)) {
+            return forbidden();
+        }
+
+        Rating savedRating = ratingService.updateRating(resultId, ratingValue);
+        return ResponseEntity.ok(savedRating);
+    }
+
+    /**
+     * Check if currently logged in user in the owner of the participation
+     *
+     * @param resultId - Id of the result that the participation belongs to
+     * @return False if User is not Owner, True otherwise
+     */
+    private boolean checkIfUserIsOwnerOfSubmission(Long resultId) {
+        User user = userService.getUser();
+        Result res = resultService.findOne(resultId);
```

Comment:
can you reduce this to one database access to find the participation directly?

Reply:
Done